src/app/views/login/SignInController.jsx

```diff
@@ -57,86 +58,121 @@ export class LoginController extends Component {
-                            console.error(errCodes.RESPONSE_ERR);
-                            notification.message = errCodes.RESPONSE_ERR;
-                            notifications.add(notification);
-                            break;
-                        }
-                        case "FETCH_ERR": {
-                            console.error(errCodes.FETCH_ERR);
-                            notification.message = errCodes.FETCH_ERR;
-                            notifications.add(notification);
-                            break;
-                        }
-                        case "UNEXPECTED_ERR": {
-                            console.error(errCodes.UNEXPECTED_ERR);
-                            notification.message = errCodes.UNEXPECTED_ERR;
-                            notifications.add(notification);
-                            break;
-                        }
-                        default: {
-                            console.error(errCodes.UNEXPECTED_ERR);
-                            notification.message = errCodes.UNEXPECTED_ERR;
-                            notifications.add(notification);
-                        }
-                    }
+                    stateToSet = this.handleLoginError(error, stateToSet);
                 }
+
+                this.setState({
+                    ...stateToSet,
+                    isSubmitting: false
+                });
+            });
+    }
+
+    handleLoginError(error, stateToSet) {
+        const notification = {
+            type: "warning",
+            isDismissable: true,
+            autoDismiss: 15000
+        };
+
+        if (error.status != null) {
+            if (error.status >= 400 && error.status < 500) {
+                let errorContents = { errorsForBody: [], emailMessage: "", passwordMessage: "" };
+                let validationErrors = {
+                    heading: "Fix the following: "
+                };
+
+                if (error.errors != null) {
+                    error.errors.forEach(anError => {
+                        errorContents = this.showValidationError(anError, errorContents, notification);
+                    });
+                } else {
+                    this.notifyUnexpectedError(notification);
+                }
+
+                validationErrors.body = errorContents.errorsForBody;
+                stateToSet = {
+                    ...this.state,
+                    validationErrors
+                };
+                stateToSet.email.errorMsg = errorContents.emailMessage;
+                stateToSet.password.errorMsg = errorContents.passwordMessage;
+            } else {
+                this.notifyUnexpectedError(notification);
+            }
+        } else {
+            this.notifyUnexpectedError(notification);
+        }
+
+        return stateToSet;
+    }
+
+    showValidationError(anError, errorContents, notification) {
+        switch (anError.code) {
+            case "InvalidEmail":
```

Comment:
Does this error mean an invalid user? If so, I'd suggest renaming to `InvalidUser` and amending the text to say something like: 'Enter a valid user email address'. 

If an error is going to be returned that handles a malformed email address, keep this `case` but have another to handle the invalid user.

Reply:
In the situation where a users account is invalid (password and email are in the correct format but no account is found), this will trigger 'NotAuthorised' error instead